src/model/liveselection.js

```diff
@@ -42,10 +42,41 @@ export default class LiveSelection extends Selection {
 		/**
 		 * Document which owns this selection.
 		 *
-		 * @private
-		 * @member {engine.model.Document} engine.model.Selection#_document
+		 * @protected
+		 * @member {engine.model.Document} engine.model.LiveSelection#_document
 		 */
 		this._document = document;
+
+		/**
+		 * Keeps mapping of attribute name to priority with which the attribute got modified (added/changed/removed)
+		 * last time. Possible values of priority are: `'low'` and `'normal'`.
+		 *
+		 * Priorities are used by internal `LiveSelection` mechanisms. All attributes set using `LiveSelection`
+		 * attributes API are set with `'normal'` priority.
+		 *
+		 * @private
+		 * @member {Map} engine.model.LiveSelection#_attributePriority
+		 */
+		this._attributePriority = new Map();
+
+		// Whenever selection range changes, if the change comes directly from selection API (direct user change).
+		this.on( 'change:range', ( evt, data ) => {
+			if ( data.directChange ) {
+				// Reset attributes on selection (clear attributes and priorities).
+				//this._resetAttributes();
```

Comment:
Te commented code is a mistake. Sorry for that.

As far as `_updateAttributes` is concerned, this issue may be resolved by simplifying `LiveSelection` internal logic and getting rid of priorities. I thought priorities, as a general solution, will be nice, but they got too clunky and even though I have those priorities, I still have to write some special code (like in `_updateAttributes`).